filters/openpolicyagent/evaluation.go

```diff
@@ -65,6 +69,46 @@ func (opa *OpenPolicyAgentInstance) Eval(ctx context.Context, req *ext_authz_v3.
 	return result, nil
 }
 
+func setDecisionIdInRequest(req *ext_authz_v3.CheckRequest, decisionId string) {
+	if metaDataContextDoesNotExist(req) {
```

Comment:
I would recommend to structure the code like 

```
if ... == nil {
  //create empty metadata
}

Fill filter metadata
```

This will reduce duplication of code ...